client/src/components/session-timer/session-timer-mixin.ts

```diff
@@ -0,0 +1,42 @@
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+
+import timerModule from '@/modules/vx-timer'
+import {
+  EXISTING_NR_TIMEOUT_MS,
+  EXISTING_NR_TIMER_NAME,
+  NR_COMPLETION_TIMEOUT_MS,
+  NR_COMPLETION_TIMER_NAME
+} from '@/store/new-request-module'
+
+import * as types from '@/store/types'
+
+@Component
+export default class SessionTimerMixin extends Vue {
+  startNewNrTimer (): Promise<any> {
+    return this.startTimer({
+      id: NR_COMPLETION_TIMER_NAME,
+      expirationFn: () => {
+        this.$store.dispatch(types.SHOW_NR_SESSION_EXPIRY_MODAL)
+          .then(() => {})
+      },
+      timeoutMs: NR_COMPLETION_TIMEOUT_MS
+    })
+  }
+
+  startExistingNrTimer (): Promise<any> {
+    return this.startTimer({
+      id: EXISTING_NR_TIMER_NAME,
+      expirationFn: () => {
+        this.$store.dispatch(types.SHOW_NR_SESSION_EXPIRY_MODAL)
+          .then(() => {})
+      },
+      timeoutMs: EXISTING_NR_TIMEOUT_MS
+    })
+  }
+
+  async startTimer (timerConfig): Promise<any> {
+    // Start the user session timer
+    return timerModule.createAndStartTimer(timerConfig)
```

Comment:
As far as I can tell, createAndStartTimer is not declared as async, so returning it isn't async, and making THIS method async doesn't do anything.

Reply:
Again, in Vue, actions are async. createAndStartTimer is an action.

Reply:
I'm genuinely trying to understand this.

Ref: https://vuex.vuejs.org/guide/actions.html#dispatching-actions-in-components

It looks like actions _can_ be async, but it looks like this one isn't.
```
export const createAndStartTimer = ({ commit }, opts: TimerOptions) => {
  commit(types.CREATE_AND_START_TIMER, opts)
}
```

Let me know if I misunderstand.

Reply:
Dispatching an action always returns a promise. I guess I didn't declare the actions async when i did the VX timer store module, but yes, dispatching those will also be async and returns a promise. Vuex does it under the hood when the action is registered.
```
export const createAndStartTimer = ({ commit }, opts: TimerOptions)
```

Reply:
It's weird Vue-ism and new to me.

This will all be a lot easier to sort through when types are implement globally (in another ticket).

OK for now.